node/src/components/block_synchronizer/block_builder/tests.rs

```diff
@@ -254,7 +254,7 @@ fn register_block_execution() {
     // execution.
     let finalized_block = Box::new(FinalizedBlock::from(block.clone()));
     builder.acquisition_state = BlockAcquisitionState::HaveFinalizedBlock(
-        *block.hash(),
+        Box::new(block),
```

Comment:
I know it's not a part of this PR, but can you update the comment, which still says about `HaveAllDeploys`.
Actually, my preference would be to remove the comment, if you don't mind :)